processing/src/main/java/io/druid/query/extraction/RegexDimExtractionFn.java

```diff
@@ -86,11 +86,8 @@ public RegexDimExtractionFn(
   @Override
   public String apply(String dimValue)
   {
-    if (dimValue == null) {
-      return null;
-    }
-    String retVal;
-    Matcher matcher = pattern.matcher(dimValue);
+    final String retVal;
+    final Matcher matcher = pattern.matcher(Strings.nullToEmpty(dimValue));
```

Comment:
The bug here was that with null inputs there are a couple cases:

- pattern matches `""` -> in this case we should return the capture group, which is definitely gonna be empty, so we'll return null
- pattern doesn't match `""` -> in this case we should return `replaceMissingValueWith` which may or may not be null

So we actually do have to check if the pattern matches `""` or not. The `testMissingValueReplacementFromNullAndEmpty` test checks for this behavior.